File: examples/documents/ros/deploy-2048/csharp/src/Demo/DemoStack.cs
using System.Collections.Generic;
using AlibabaCloud.SDK.ROS.CDK.Core;
using AlibabaCloud.SDK.ROS.CDK.Ecs;

namespace Demo
{
    public class DemoStack : Stack
    {
        public DemoStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
        {
            // The code that defines your stack goes here
            new RosInfo(this, RosInfo.Description, "Deploy 2048 game through ROS CDK.");

            var zoneId = new RosParameter(this, "ZoneId", new RosParameterProps{
                Type = RosParameterType.STRING
            });

            var vpc = new Vpc(this, "Vpc", new VPCProps{
                CidrBlock = "192.168.0.0/16"
            });

            var sg = new SecurityGroup(this, "SecurityGroup", new SecurityGroupProps{
                VpcId = vpc.Ref
            });

            var vsw = new VSwitch(this, "VSwitch", new VSwitchProps{
                CidrBlock = "192.168.0.0/24",
                ZoneId = zoneId.Value,
                VpcId = vpc.Ref
            });

            var instance = new Instance(this, "Instance", new InstanceProps{
                VpcId = vpc.Ref,
                VSwitchId = vsw.Ref,
                SecurityGroupId = sg.Ref,
                ZoneId = zoneId.Value,
                InstanceType = "ecs.c6e.large",
                ImageId = "centos_7_8",
                SystemDiskCategory = "cloud_essd",
                IoOptimized = "optimized"
            });

            var sgIngress = new SecurityGroupIngress(this, "SecurityGroupIngress", new SecurityGroupIngressProps{
                IpProtocol = "tcp",
                PortRange = "80/80",
                SourceCidrIp = "0.0.0.0/0",
                SecurityGroupId = sg.Ref,
                NicType = "intranet"
            });

            var runCommand = new RunCommand(this, "RunCommand", new RunCommandProps{
                InstanceIds = new IResolvable[] { instance.AttrInstanceId },
                CommandContent = "yum install -y httpd wget\nsystemctl start httpd\nwget 'https://computenest-artifacts-cn-hangzhou.oss-cn-hangzhou-internal.aliyuncs.com/1853370294850618/cn-beijing/1697533575326/2048.tgz' -O 2048.tgz\ntar xvf 2048.tgz\nmv 2048/* /var/www/html && rm -rf 2048",
                Type = "RunShellScript",
                Timeout = 3600,
                Sync = true
            });
            runCommand.AddDependency(sgIngress);

            new RosOutput(this, "Url", new RosOutputProps{
                Value = "http://" + instance.AttrPublicIp.ToString() + ":80"
            });
        }
    }
}

File: examples/documents/ros/deploy-2048/python/demo/demo_stack.py
import ros_cdk_core as core
import ros_cdk_ecs as ecs


class DemoStack(core.Stack):

    def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        # The code that defines your stack goes here
        core.RosInfo(self, core.RosInfo.description, "Deploy 2048 game through ROS CDK.")

        zone_id = core.RosParameter(self, "ZoneId", type=core.RosParameterType.STRING)

        vpc = ecs.Vpc(self, "Vpc", ecs.VPCProps(cidr_block="192.168.0.0/16"))

        sg = ecs.SecurityGroup(self, "SecurityGroup", ecs.SecurityGroupProps(vpc_id=vpc.ref))

        vsw = ecs.VSwitch(self, "VSwitch", ecs.VSwitchProps(
            cidr_block='192.168.0.0/24',
            vpc_id=vpc.ref,
            zone_id=zone_id,
        ))

        instance = ecs.Instance(self, "Instance", ecs.InstanceProps(
            vpc_id=vpc.ref,
            instance_type="ecs.c6e.large",
            image_id="centos_7_8",
            security_group_id=sg.ref,
            v_switch_id=vsw.ref,
            zone_id=zone_id,
            system_disk_category="cloud_essd",
            io_optimized="optimized"
        ))

        sg_ingress = ecs.SecurityGroupIngress(self, "SecurityGroupIngress", ecs.SecurityGroupIngressProps(
            security_group_id=sg.ref,
            port_range="80/80",
            ip_protocol="tcp",
            source_cidr_ip="0.0.0.0/0",
            nic_type="intranet"
        ))

        run_command = ecs.RunCommand(self, "RunCommand", ecs.RunCommandProps(
            instance_ids=[instance.ref],
            command_content= "yum install -y httpd wget\nsystemctl start httpd\nwget 'https://computenest-artifacts-cn-hangzhou.oss-cn-hangzhou-internal.aliyuncs.com/1853370294850618/cn-beijing/1697533575326/2048.tgz' -O 2048.tgz\ntar xvf 2048.tgz\nmv 2048/* /var/www/html && rm -rf 2048",
            type="RunShellScript",
            timeout=3600,
            sync=True
        ))
        run_command.add_dependency(sg_ingress)

        core.RosOutput(self, "Url", value="http://{}:80".format(instance.attr_public_ip.to_string()))
